packages/neon_framework/lib/src/storage/sqlite_cookie_persistence.dart

```diff
@@ -0,0 +1,296 @@
+            if (isHttpRequest) 1 else 0,
+          ],
+        );
+
+        await txn.rawUpdate(
+          '''
+UPDATE "cookies"
+SET "last-access-time" = datetime('now', 'utc')
+WHERE "account" = ?
+  AND ("domain" = ? OR ("host-only-flag" = 0 AND "domain" LIKE '%.' || ?))
+  AND "path" LIKE ? || '%'
+  AND ("secure-only-flag" = 0 OR ? = 1)
+  AND ("http-only-flag" = 0 OR ? = 1)
+  AND datetime("expiry-time", 'unixepoch', 'utc') <= datetime('now', 'utc')
+''',
+          [
+            accountID,
+            uri.host,
+            uri.host,
+            path,
+            if (isSecureRequest) 1 else 0,
+            if (isHttpRequest) 1 else 0,
+          ],
+        );
+
+        return results;
+      });
+
+      for (final result in results) {
+        final cookie = Cookie(
+          result['name']! as String,
+          result['value']! as String,
+        );
+
+        list.add(cookie);
+      }
+    } on DatabaseException catch (error, stackTrace) {
+      _log.warning(
+        'Error loading cookies.',
+        error,
+        stackTrace,
+      );
+    }
+
+    return list;
+  }
+
+  @override
+  Future<void> saveFromResponse(Set<StorableCookie> cookies, {required bool isHttpRequest}) async {
+    final batch = _requireDatabase.batch();
+
+    for (final cookie in cookies) {
+      if (allowedBaseUri != null) {
+        if (!isDomainMatch(allowedBaseUri!.host, cookie.domain) || !isPathMatch(allowedBaseUri!.path, cookie.path)) {
+          continue;
+        }
+      }
+
+      // Always store paths as folders for easier lookup.
+      var path = cookie.path;
+      if (!path.endsWith('/')) {
+        path += '/';
+      }
+
+      batch
+        ..update(
+          'cookies',
+          {
+            '"name"': cookie.name,
+            '"value"': cookie.value,
+            '"expiry-time"': cookie.expiryTime.secondsSinceEpoch,
+            '"domain"': cookie.domain,
+            '"path"': path,
+            // DO NOT update the creation time
+            '"last-access-time"': cookie.lastAccessTime.secondsSinceEpoch,
+            '"persistent-flag"': cookie.persistentFlag ? 1 : 0,
+            '"host-only-flag"': cookie.hostOnlyFlag ? 1 : 0,
+            '"secure-only-flag"': cookie.secureOnlyFlag ? 1 : 0,
+            '"http-only-flag"': cookie.httpOnlyFlag ? 1 : 0,
+          },
+          where:
+              '"account" = ? AND "name" = ? AND "domain" = ? AND "path" = ? ${!isHttpRequest ? '"AND http-only-flag" = 1' : ''}',
+          whereArgs: [accountID, cookie.name, cookie.domain, cookie.path],
+        )
+        // Ignore already present records.
+        // This happens when the cookie was not updated because of the http-only-flag
+        ..rawInsert(
+          '''
+INSERT OR IGNORE INTO cookies ("account", "name", "value", "expiry-time", "domain", "path", "creation-time", "last-access-time", "persistent-flag", "host-only-flag", "secure-only-flag", "http-only-flag")
+SELECT ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?
```

Comment:
Why SELECT and not VALUES?

Reply:
Because a normal insert does not support the `WHERE` clause.
It is the same pattern we use for other SQLite persistences.

Reply:
Ah right I didn't see the next line.